source/agora/consensus/ValidatorSet.d

```diff
@@ -562,6 +563,46 @@ public class ValidatorSet
 
         return true;
     }
+
+    /***************************************************************************
+
+        Restore validators' information from blocks
+
+        Params:
+            blocks = blocks to get validators' information
+            finder = the delegate to find UTXOs with
+
+        Returns:
+            true if the restoring information was successful, false otherwise
+
+    ***************************************************************************/
+
+    public bool restoreValidators (const(Block)[] blocks,
+        scope UTXOFinder finder) @safe nothrow
+    {
+        foreach (const ref block; blocks)
+        {
+            if (block.header.enrollments.length > 0)
+            {
+                foreach (const ref enroll; block.header.enrollments)
+                {
+                    try
+                    {
+                        this.add(block.header.height, finder, enroll);
```

Comment:
You are wrapping these functions in a try/catch even though they never throw. And you are **ignoring** the return values which are booleans.

And now you introduced yet another `bool` function `restoreValidators`.

Do you understand the problem with bool return functions now?

Reply:
Yes, I do. Sorry for that. I will deal with it.